pkg/providers/vault/config.go

```diff
@@ -17,18 +17,21 @@ package vault
 
 import (
 	"fmt"
+	"time"
 )
 
 // SecretConfig contains configuration settings used to communicate with an HTTP based secret provider
 type SecretConfig struct {
-	Host           string
-	Port           int
-	Path           string
-	Protocol       string
-	Namespace      string
-	RootCaCertPath string
-	ServerName     string
-	Authentication AuthenticationInfo
+	Host                    string
+	Port                    int
+	Path                    string
+	Protocol                string
+	Namespace               string
+	RootCaCertPath          string
+	ServerName              string
+	Authentication          AuthenticationInfo
+	AdditionalRetryAttempts int
+	RetryWaitPeriod         time.Duration
```

Comment:
During my testing while integrating this with `edgex-go` this configuration was being treated as a `nanosecond` value. For example, I provided :
```
...
AdditionalRetryAttempts = 5
RetryWaitPeriod = 10
...
```

Is this intended or is my configuration values incorrect?

Reply:
you need to pass a time.Duration including the units, so if you wanted 10 nanoseconds, the "right" way to do this would be:

```
RetryWaitPeriod = 10*time.Nanosecond
```
or if you wanted 10 seconds, then you would do

```
RetryWaitPeriod = 10*time.Second
```

See https://play.golang.org/p/Uo7UP9SwrOj for an example

it is kinda unfortunate that time.Duration is really just type-cast as an int and internally stored as nanoseconds (with the time.Second, etc. constants just being defined as the number of nanoseconds per that unit, so time.Second becomes 1000000000), because it means specifying just raw numbers is accepted blindly as nanoseconds which is almost always what folks _don't_ mean. OTOH it does make arithmetic with time.Duration quite easy :man_shrugging: 

TLDR, yes you're holding it wrong :smile:

Reply:
How would we specify that in a configuration TOML? `edgex-go` basically passed the `SecretConfig` values though to `go-mod-secrets` without any additional processing.

Reply:
Oh I see... you want to specify this through configuration.toml.

What I'd recommend there instead is to just have the RetryWaitPeriod in configuration.toml be a time string, parse it into a time.Duration with [`ParseDuration`](https://golang.org/pkg/time/#ParseDuration)

We could accept a string here and do the parsing inside go-mod-secrets but I don't like doing that because having a time.Duration exposed from a Go interface perspective is definitely the right:tm: thing to do from a Go programming perspective, but if that's what you'd prefer I can refactor to do that as well I suppose.